documentation/modules/ROOT/pages/connectors/mongodb.adoc

```diff
@@ -95,6 +95,13 @@ In this way the connector dynamically adjusts to changes in replica set membersh
 * link:https://docs.mongodb.com/manual/tutorial/remove-shards-from-cluster/[Shard removal]
 * link:https://docs.mongodb.com/manual/changeStreams/[Change Streams]
 
+// Type: concept
+// Title: Description of how the MongoDB connector uses read preference
+// ModuleID: description-of-how-the-mongodb-connector-uses-read-preference
+[id="read-preference"]
+=== Read Preference
+Starting with Debezium 2.4 the MongoDB connector honors the link:https://www.mongodb.com/docs/manual/core/read-preference/[read preference] configured in either xref:mongodb-property-mongodb-connection-string[`mongodb.connection.string`] or xref:mongodb-property-mongodb-connection-string-shard-params[`mongodb.connection.string.shard.params`] for sharded clusters when xref:mongodb-property-mongodb-connection-mode[`mongodb.connection.mode`] is set to `replica_set`.
```

Comment:
```suggestion
Beginning with Debezium 2.4, when the MongoDB connector captures changes from a sharded cluster, and the xref:mongodb-property-mongodb-connection-mode[`mongodb.connection.mode`] is set to `replica_set`, the connector honors the link:https://www.mongodb.com/docs/manual/core/read-preference/[read preference] specified in either the xref:mongodb-property-mongodb-connection-string[`mongodb.connection.string`] or the xref:mongodb-property-mongodb-connection-string-shard-params[`mongodb.connection.string.shard.params`].
```

Reply:
@roldanbob  I believe this isn't right. Your wording changes the meaning. The important note here is that `mongodb.connection.string` is used for everything but sharded cluster  with `mongodb.connection.mode` set to `replica_set`. In such case `mongodb.connection.string.shard.params` is used instead

Reply:
Thanks, @jcechace I think that I've addressed the issue. Please take a look to verify.

Reply:
@roldanbob  not quite... though reading my original post it is ambiguous.

So.. By default in 2.4 the connector honors the setting in `mongodb.connection.string`. However specifically (and only) for sharded clusters with `mongodb.connection.mode` set to `replica_set` the setting is taken from `mongodb.connection.string.shard.params`.

This is because for all other combinations of connnector config and MongoDB topology the connector uses the connection String set in  `mognodb.connection.string` to connect to the database AND capture changes. However, for sharded clusters with `mongodb.connection.mode` set to `replica_set` (as opposed to `shared`) the initial connection string from `mongodb.connection.string` is used ONLY to read the cluster topology from the `config.shards` collection. Once that is done a new connection string is constructed for each shard by combining the information retrieved from `config.shards` collection and the parameters specified in `mongodb.connection.string.shard.pamars`.

Reply:
Ok, yeah, thanks for teasing that out a little more.  I was/am still a little confused about the linkage between how Debezium connects to a shard, and the read preference mode, but it's beginning to make sense. I guess that the read preference is implied from the connection interface? 
 I'm not sure that I've cracked this nut yet, but here's another go at it. Getting closer?